lib/ansible/plugins/strategy/__init__.py

```diff
@@ -874,7 +872,9 @@ def _do_handler_run(self, handler, handler_name, iterator, play_context, notifie
         run_once = False
         try:
             action = action_loader.get(handler.action, class_only=True)
-            if handler.run_once or getattr(action, 'BYPASS_HOST_LOOP', False):
+            handler_vars = self._variable_manager.get_vars(play=iterator._play, task=handler)
+            templar = Templar(loader=self._loader, variables=handler_vars)
```

Comment:
instead of forcing templating here, when you don't have task_vars yet, just set run_once based on 'bypass' and then template below in 890 with full task_vars.

Reply:
ah, missed that. Thanks!